packages/ra-ui-materialui/src/input/ArrayInput/SimpleFormIterator.tsx

```diff
@@ -57,13 +59,22 @@ export const SimpleFormIterator = (props: SimpleFormIteratorProps) => {
         [remove]
     );
 
+    if (fields.length > 0) {
+        const { id, ...rest } = fields[0];
+        initialDefaultValue = rest;
+        for (const k in initialDefaultValue) initialDefaultValue[k] = '';
+    }
+
     const addField = useCallback(
         (item: any = undefined) => {
             let defaultValue = item;
             if (item == null) {
+                defaultValue = initialDefaultValue;
                 if (
                     Children.count(children) === 1 &&
                     React.isValidElement(Children.only(children)) &&
+                    (Children.only(children) as ReactElement<any>).type !==
```

Comment:
Can we avoid inspecting the child type?

Reply:
Actually, I just realized that check is not necessary there at all